addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js

```diff
@@ -248,15 +248,7 @@ export class TicketScreen extends Component {
             return;
         }
 
-        // The order that will contain the refund orderlines.
-        // Use the destinationOrder from props if the order to refund has the same
-        // partner as the destinationOrder.
-        const destinationOrder =
-            this.props.destinationOrder &&
-            partner === this.props.destinationOrder.get_partner() &&
-            !this.pos.doNotAllowRefundAndSales()
-                ? this.props.destinationOrder
-                : this._getEmptyOrder(partner);
+        const destinationOrder = this._getEmptyOrder(partner);
```

Comment:
Can you tell me why you changed this ?

Reply:
the idea here is that if you are refunding an order, you will not have anything else in that order. That means that the `destinationOrder` always corresponds to a new order

Reply:
i noticed that i didn't account for all the usecases before. I made a new change now

Reply:
> the idea here is that if you are refunding an order, you will not have anything else in that order. That means that the `destinationOrder` always corresponds to a new order

I'm not sure about this. The task tells that there must not be multiple refunds in a single order but there might be other sales in the same order. For me the solution was to add a function to look if the destinationOrder already had refund orderlines of another past order. If not and if all the other function were ok we could use the destinationOrder. I admit this is a bit more messy in the code but this lead to people coming at the cashier, wanting to buy things and telling at the same time "oh I also want to return this". They will have only 1 order for everything which is great. Tell me what you think of this ?

Reply:
You are right, but at some point we were saying that it would be very convenient to only have refund items in a refund order. What do you think, @caburj ? Should we make is so a refund order contains only refunds and a regular order contains no refunds? 

If we have to keep this functionality, i'll use @adgu-odoo 's solution

Reply:
I think that is more a question for @dama-odoo as it is more functional than technical

Reply:
we'll keep the functionality.